airflow/contrib/operators/gcp_vision_operator.py

```diff
@@ -0,0 +1,603 @@
+    :param timeout: (Optional) The amount of time, in seconds, to wait for the request to
+        complete. Note that if retry is specified, the timeout applies to each individual
+        attempt.
+    :type timeout: float
+    :param metadata: (Optional) Additional metadata that is provided to the method.
+    :type metadata: Sequence[Tuple[str, str]]
+    :param gcp_conn_id: The connection ID used to connect to Google Cloud Platform.
+    :type gcp_conn_id: str
+
+    """
+    # [START vision_product_get_template_fields]
+    template_fields = ('location', 'project_id', 'product_id', 'gcp_conn_id')
+    # [END vision_product_get_template_fields]
+
+    @apply_defaults
+    def __init__(self,
+                 location,
+                 product_id,
+                 project_id=None,
+                 retry=None,
+                 timeout=None,
+                 metadata=None,
+                 gcp_conn_id='google_cloud_default',
+                 *args, **kwargs):
+        super(CloudVisionProductGetOperator, self).__init__(*args, **kwargs)
+        self.location = location
+        self.product_id = product_id
+        self.project_id = project_id
+        self.retry = retry
+        self.timeout = timeout
+        self.metadata = metadata
+        self.gcp_conn_id = gcp_conn_id
+        self._hook = CloudVisionHook(gcp_conn_id=self.gcp_conn_id)
+
+    def execute(self, context):
+        return self._hook.get_product(location=self.location,
+                                      product_id=self.product_id,
+                                      project_id=self.project_id,
+                                      retry=self.retry,
+                                      timeout=self.timeout,
+                                      metadata=self.metadata)
+
+
+class CloudVisionProductUpdateOperator(BaseOperator):
+    """
+    Makes changes to a Product resource. Only the display_name, description, and labels fields can be
+    updated right now.
+
+    If labels are updated, the change will not be reflected in queries until the next index time.
+
+    Note:
+
+    To locate the `Product` resource, its `name` in the form
+    `projects/PROJECT_ID/locations/LOC_ID/products/PRODUCT_ID` is necessary.
+
+    You can provide the `name` directly as an attribute of the `product` object. However, you can leave it
+    blank and provide `location` and `product_id` instead (and optionally `project_id` - if not present,
+    the connection default will be used) and the `name` will be created by the operator itself.
+
+    This mechanism exists for your convenience, to allow leaving the `project_id` empty and having Airflow
+    use the connection default `project_id`.
+
+    Possible errors related to the provided `Product`:
+
+    - Returns NOT_FOUND if the Product does not exist.
+    - Returns INVALID_ARGUMENT if display_name is present in update_mask but is missing from the request or
+    longer than 4096 characters.
```

Comment:
```suggestion
        longer than 4096 characters.
```
See: http://docutils.sourceforge.net/docs/user/rst/quickref.html#bullet-lists

Reply:
Done